javascript/implementation/KeySet.ts

```diff
@@ -0,0 +1,228 @@
+     * @param asOf
+     * @returns an async iterator across everything in the key set, with values returned as pairs of Muid, Key
+     */
+    entries(asOf?: AsOf): AsyncGenerator<[Muid,KeyType], void, unknown> {
+        const thisSet = this;
+        return (async function*(){
+            const entries = await thisSet.ginkInstance.store.getKeyedEntries(thisSet.address, asOf);
+            for (const [key, entry] of entries) {
+                yield [muidTupleToMuid(entry.entryId), key]
+            }
+        })();
+    }
+
+    /**
+     * Returns whether the key set has a key or not.
+     * @param key
+     * @param asOf
+     * @returns true if the key set has the key, false if not.
+     */
+    async has(key: KeyType, asOf?: AsOf): Promise<boolean> {
+        const result = await this.ginkInstance.store.getEntryByKey(this.address, key, asOf);
+        return result != undefined;
+    }
+
+    /**
+     * Returns the contents of the key set as a set.
+     * @param asOf
+     * @returns a promise that resolves to a set with KeyTypes.
+     */
+    async toSet(asOf?: AsOf): Promise<Set<KeyType>> {
+        const entries = await this.ginkInstance.store.getKeyedEntries(this.address, asOf);
+        const resultSet = new Set<KeyType>;
+        for (const [key, entry] of entries) {
+            resultSet.add(key);
+        }
+        return resultSet;
+    }
+
+    /**
+     * How many entries are in the key set.
+     * @param asOf
+     * @returns a promise that resolves to a number.
+     */
+    async size(asOf?: AsOf): Promise<number> {
+        const entries = await this.ginkInstance.store.getKeyedEntries(this.address, asOf);
+        return entries.size;
+    }
+
+    /**
+     * Returns a boolean stating whether all contents of the specified set are in the key set
+     * @param subset another set of KeyTypes to compare contents to
+     * @returns a promise resolving to true if every element from subset are in the key set
+     */
+    async isSuperset(subset: Iterable<KeyType>): Promise<boolean> {
+        for (const elem of subset) {
+            if (!await this.has(elem)) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    /**
+     * All values in either the key set or the provided iterable
+     * @param iterable an iterable of KeyTypes
+     * @returns a promise resolving to a set of KeyTypes
+     */
+    async union(iterable: Iterable<KeyType>): Promise<Set<KeyType>> {
+        const _union = await this.toSet();
+        for (const elem of iterable) {
+            await _union.add(elem);
+        }
+        return _union;
+    }
+
+    /**
+     * All values that are in both the key set and the provided iterable.
+     * @param iterable an iterable of KeyTypes
+     * @returns a promise resolving to a set of KeyTypes
+     */
+    async intersection(iterable: Iterable<KeyType>): Promise<Set<KeyType>> {
```

Comment:
Also, as with union, etc. it would be better to leave this off since people who have used the Javascript Set would not expect to have this, and simpler is better.